File: leetcode/415. Add Strings/415.cpp
// https://helloacm.com/how-to-add-two-strings-numerically/
// https://leetcode.com/problems/add-strings/

class Solution {
public:
    string addStrings(string num1, string num2) {
        string r = "";
        int sz1 = num1.size();
        int sz2 = num2.size();
        int i = 0, j = 0, c = 0;
        while (i < sz1 || j < sz2 || c > 0) {
            int t = c;
            t += i < sz1 ? num1[sz1 - i - 1] - 48 : 0;
            t += j < sz2 ? num2[sz2 - j - 1] - 48 : 0;
            r = string(1, t % 10 + 48) + r;
            c = t / 10;
            i ++;
            j ++;
        }
        return r;
    }
};

File: leetcode/415. Add Strings/415.py
# https://helloacm.com/teaching-kids-programming-add-two-big-integers-in-strings/
# https://leetcode.com/problems/add-strings/
# EASY, MATH

class Solution:
    def addStrings(self, num1: str, num2: str) -> str:
        n1, n2 = len(num1) - 1, len(num2) - 1
        ans = deque()
        c = 0
        while n1>=0 or n2>=0 or c>0:
            x = c
            if n1>=0:
                x += int(num1[n1])
                n1 -= 1
            if n2>=0:
                x += int(num2[n2])
                n2 -= 1
            ans.appendleft(str(x%10))
            c = x // 10
        return "".join(ans)
